docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -269,10 +265,11 @@ public Node call() throws Exception {
                                     //
                                     // deferring the completion of provisioning until the launch
```

Comment:
I am not super familiar with the cloud api, but @stephenc please tell me if I am understanding.

NodeProvisioner.PlannedNode takes a Future. Currently, we provision a slave (including doing some bad things) and return a node immediately despite it not being connected yet. Instead what we should be doing is creating a slave and get connected to the docker instance and finally resolve the future.

I'm guessing if that is what were are supposed to be doing there is some gotcha that prevents us from creating a slave that way...